core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+      if (processor.announceArc) {
+        if (!reverse)
+          ray.direction.scaleInPlace(-1);
+        processor.announceArc(arcBetween2Samples(ray, p0, true), f0, f1);
+      }
+      if (processor.announceChord)
+        processor.announceChord(p0, ray.origin, f0, f1);
+    };
+    const reverseFractionsForSymmetry = (q: QuadrantFractions): boolean => {
+      // if we touch an axis, start at the major axis so the approximation exhibits axial symmetry
+      if (!q.axisAtStart && !q.axisAtEnd)
+        return false;
+      const xAxisIsMajor = ellipticalArc.vector0.magnitudeSquared() > ellipticalArc.vector90.magnitudeSquared();
+      const reverse = xAxisIsMajor ? (q.quadrant === 2 || q.quadrant === 4) : (q.quadrant === 1 || q.quadrant === 3);
+      if (reverse)
+        q.fractions.reverse();
+      return reverse;
+    };
+
+    for (const q of samples) {
+      const n = q.fractions.length;
+      if (n < 2 || !processor.announceQuadrantBegin(q))
+        continue;
+      const reverse = reverseFractionsForSymmetry(q);
+      let pts = createFirstArc(q.fractions[0], q.fractions[1], reverse);
+      for (let i = 2; i < n - 1; ++i) // need at least 4 samples to make inner arcs
+        pts = createInnerArc(pts.p0, pts.p1, q.fractions[i - 1], q.fractions[i]);
+      if (n > 2)
+        createLastArc(pts.p2, q.fractions[n - 2], q.fractions[n - 1], reverse);
+      if (reverse)
+        q.fractions.reverse();  // restore original order just to be safe
+      processor.announceQuadrantEnd(q, reverse);
+    }
+  }
+  /**
+   * Compute the maximum error of the given approximation.
+   * * This is measured by the longest perpendicular between the elliptical arc and its approximation.
+   * @param samples structured sample data for the instance's elliptical arc.
+   * @param arcChain whether to compute approximation error of a circular arc chain (true) or linestring (false)
+   * @return details of the perpendicular measuring the max approximation error, or undefined if no such perpendicular.
+   * For each of `detailA` and `detailB`:
+   * * `point` is the end of the perpendicular on each curve
+   * * `fraction` is the curve parameter of the point
+   * * `a` is the distance between the points.
+   */
+  public computeApproximationError(samples: QuadrantFractions[], arcChain: boolean): CurveLocationDetailPair | undefined {
+    if (!this.isValidArc)
+      return undefined;
+    const arcXY = this.cloneLocalArc();
+    const processor = arcChain ? new ArcChainErrorProcessor(arcXY) : new LineStringErrorProcessor(arcXY);
+    EllipticalArcApproximationContext.processQuadrantFractions(arcXY, samples, processor);
+    const maxError = processor.maxPerpendicular;
+    return (maxError && maxError.tryTransformInPlace(this.localToWorld)) ? maxError : undefined;
+  }
+  /**
+   * Compute samples for the elliptical arc as fraction parameters.
+   * @param options approximation options
+   * @param structuredOutput
```

Comment:
Please add what `structuredOutput` is

Reply:
oh you have its details on the next line. Still if you add couple words line `a flag for whatever` makes the doc more readable.